### PR TITLE
Copy test suite properties when exploring with a filter

### LINKED ISSUE #3611: Properties are shown when --explore:nunit3 is run on entire project, but omitted when using the --where clause
I am trying to get all the tests that contain a specific category, but I still want to know all the other categories associated to those test fixtures and test cases.

I am using NUnit.ConsoleRunner v3.8.0

If I run 
nunit3-console.exe "C:\Users\SWycoff\Documents\GitHub\myon_qa_suite\myON\myON\bin\Debug\myON.dll" --explore:nunit3

I get test fixtures and test cases with their properties like you see below:

```
<?xml

 version="1.0" encoding="utf-8" standalone="no"?>
<test-run id="2" testcasecount="2715">
  <test-suite type="Assembly" id="0-5019" name="myON.dll" fullname="C:\Users\SWycoff\Documents\GitHub\myon_qa_suite\myON\myON\bin\Debug\myON.dll" runstate="Runnable" testcasecount="2715">
    <properties>
      <property name="LevelOfParallelism" value="1" />
      <property name="_PID" value="27168" />
      <property name="_APPDOMAIN" value="domain-" />
    </properties>
    <test-suite type="TestSuite" id="0-5020" name="myON" fullname="myON" runstate="Runnable" testcasecount="2715">
      <test-suite type="TestFixture" id="0-1000" name="BA_CreateProject_AllTaskTypes_Books" fullname="myON.BA_CreateProject_AllTaskTypes_Books" classname="myON.BA_CreateProject_AllTaskTypes_Books" runstate="Runnable" testcasecount="1">
        <properties>
          <property name="ParallelScope" value="Self" />
          <property name="Category" value="ba" />
          <property name="Category" value="appareaprojects" />
          <property name="Category" value="permissionbooks4core" />
        </properties>
        <test-suite type="ParameterizedMethod" id="0-1002" name="BA030063_CreateAProject_AllAvailTasks_Books" fullname="myON.BA_CreateProject_AllTaskTypes_Books.BA030063_CreateAProject_AllAvailTasks_Books" classname="myON.BA_CreateProject_AllTaskTypes_Books" runstate="Runnable" testcasecount="1">
          <properties>
            <property name="Category" value="priority2" />
            <property name="Description" value="Creates a project task of each available type, verifies that the tasks save." />
          </properties>
          <test-case id="0-1001" name="BA030063_CreateAProject_AllAvailTasks_Books()" fullname="myON.BA_CreateProject_AllTaskTypes_Books.BA030063_CreateAProject_AllAvailTasks_Books()" methodname="BA030063_CreateAProject_AllAvailTasks_Books" classname="myON.BA_CreateProject_AllTaskTypes_Books" runstate="Runnable" seed="2012297899" />
        </test-suite>
      </test-suite>
      <test-suite type="TestFixture" id="0-1003" name="BA_CreateProject_AllTaskTypes_Reader" fullname="myON.BA_CreateProject_AllTaskTypes_Reader" classname="myON.BA_CreateProject_AllTaskTypes_Reader" runstate="Runnable" testcasecount="1">
        <properties>
          <property name="ParallelScope" value="Self" />
          <property name="Category" value="ba" />
          <property name="Category" value="appareaprojects" />
          <property name="Category" value="permissionreader4corepluslexile" />
        </properties>
        <test-suite type="ParameterizedMethod" id="0-1005" name="BA030064_CreateAProject_AllAvailTasks_Reader" fullname="myON.BA_CreateProject_AllTaskTypes_Reader.BA030064_CreateAProject_AllAvailTasks_Reader" classname="myON.BA_CreateProject_AllTaskTypes_Reader" runstate="Runnable" testcasecount="1">
          <properties>
            <property name="Category" value="priority1" />
            <property name="Description" value="Creates a project task of each available type, verifies that the tasks save." />
          </properties>
          <test-case id="0-1004" name="BA030064_CreateAProject_AllAvailTasks_Reader()" fullname="myON.BA_CreateProject_AllTaskTypes_Reader.BA030064_CreateAProject_AllAvailTasks_Reader()" methodname="BA030064_CreateAProject_AllAvailTasks_Reader" classname="myON.BA_CreateProject_AllTaskTypes_Reader" runstate="Runnable" seed="1186221310" />
        </test-suite>
      </test-suite>

```

But if I add any where clauses , for example run this:
nunit3-console.exe "C:\Users\SWycoff\Documents\GitHub\myon_qa_suite\myON\myON\bin\Debug\myON.dll" --explore:nunit3 --where "cat == priority4"

I get the two test cases, but no properties on the fixture or the test case itself.

```
<?xml version="1.0" encoding="utf-8" standalone="no"?>
<test-run id="2" testcasecount="2">
  <test-suite type="Assembly" id="0-5021" name="myON.dll" fullname="C:\Users\SWycoff\Documents\GitHub\myon_qa_suite\myON\myON\bin\Debug\myON.dll" runstate="Runnable" testcasecount="2">
    <test-suite type="TestSuite" id="0-5022" name="myON" fullname="myON" runstate="Runnable" testcasecount="2">
      <test-suite type="TestSuite" id="0-5023" name="SA_Login_NoPermissions" fullname="myON.SA_Login_NoPermissions" runstate="Runnable" testcasecount="2">
        <test-suite type="TestSuite" id="0-5024" name="SA000001D_Login_NoPermissions_LLGradeRandom" fullname="myON.SA_Login_NoPermissions.SA000001D_Login_NoPermissions_LLGradeRandom" classname="myON.SA_Login_NoPermissions" runstate="Runnable" testcasecount="1">
          <test-case id="0-2425" name="SA000001D_Login_NoPermissions_LLGradeRandom()" fullname="myON.SA_Login_NoPermissions.SA000001D_Login_NoPermissions_LLGradeRandom()" methodname="SA000001D_Login_NoPermissions_LLGradeRandom" classname="myON.SA_Login_NoPermissions" runstate="Runnable" seed="862805472">
            <properties />
          </test-case>
        </test-suite>
        <test-suite type="TestSuite" id="0-5025" name="SA000001E_Login_NoPermissions_ULGradeRandom" fullname="myON.SA_Login_NoPermissions.SA000001E_Login_NoPermissions_ULGradeRandom" classname="myON.SA_Login_NoPermissions" runstate="Runnable" testcasecount="1">
          <test-case id="0-2427" name="SA000001E_Login_NoPermissions_ULGradeRandom()" fullname="myON.SA_Login_NoPermissions.SA000001E_Login_NoPermissions_ULGradeRandom()" methodname="SA000001E_Login_NoPermissions_ULGradeRandom" classname="myON.SA_Login_NoPermissions" runstate="Runnable" seed="591938777">
            <properties />
          </test-case>
        </test-suite>
      </test-suite>
    </test-suite>
  </test-suite>
</test-run>

```

This was originally posted in google groups.  Here is the conversation from that.

Charlie said "That seems wrong to me. It looks as if the implementation of the fix for filtered explore output may be using different code from that which creates the output for the unfiltered case. I don't even see the fixture itself (no type="TestFixture")."

### PR DESCRIPTION
**Pull request by cjconrad2**
Fixes #3611 

I've added a test that compares properties between a TestFixture from an ITestAssemblyRunner to the same fixture retrieved via ExploreTests, but would appreciate any feedback around the right way to cover this. 

I've manually tested this against a test assembly I have from another project. Ran explore with `--where 'cat=AppFamilyUtilitiesTests`.

Before changes, result was:
```xml
<?xml version="1.0" encoding="utf-8" standalone="no"?>
<test-run id="0" runstate="Runnable" testcasecount="10">
  <test-suite type="Assembly" id="0-1163" name="Foreman.Tests.dll" fullname="C:\dev\stash\devops\jobbuilder\JobBuilder\Foreman.Tests\bin\Debug\Foreman.Tests.dll" runstate="Runnable" testcasecount="10">
    <test-suite type="TestSuite" id="0-1164" name="Foreman" fullname="Foreman" runstate="Runnable" testcasecount="10">
      <test-suite type="TestSuite" id="0-1165" name="Tests" fullname="Foreman.Tests" runstate="Runnable" testcasecount="10">
        <test-suite type="TestSuite" id="0-1166" name="HelpersTest" fullname="Foreman.Tests.HelpersTest" runstate="Runnable" testcasecount="10">
          <test-suite type="TestFixture" id="0-1167" name="AppFamilyUtilitiesTests" fullname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests" runstate="Runnable" testcasecount="10">
            <test-suite type="ParameterizedMethod" id="0-1169" name="GetRoleIdToAppFamilyToADGroupCNTests" fullname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests.GetRoleIdToAppFamilyToADGroupCNTests" classname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests" runstate="Runnable" testcasecount="2">
              <test-case id="0-1090" name="Empty AppFamilyGroupModel list" fullname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests.Empty AppFamilyGroupModel list" methodname="GetRoleIdToAppFamilyToADGroupCNTests" classname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests" runstate="Runnable" seed="1242564091">
                <properties />
              </test-case>
              <test-case id="0-1091" name="Three roles and with multiple app families per role" fullname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests.Three roles and with multiple app families per role" methodname="GetRoleIdToAppFamilyToADGroupCNTests" classname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests" runstate="Runnable" seed="1740957023">
                <properties />
              </test-case>
            </test-suite>
          </test-suite>
        </test-suite>
      </test-suite>
    </test-suite>
  </test-suite>
</test-run>
```

After changes:
```xml
<?xml version="1.0" encoding="utf-8" standalone="no"?>
<test-run id="0" runstate="Runnable" testcasecount="10">
  <test-suite type="Assembly" id="0-1163" name="Foreman.Tests.dll" fullname="C:\dev\stash\devops\jobbuilder\JobBuilder\Foreman.Tests\bin\Debug\Foreman.Tests.dll" runstate="Runnable" testcasecount="10">
    <properties>
      <property name="_PID" value="4100" />
      <property name="_APPDOMAIN" value="domain-" />
    </properties>
    <test-suite type="TestSuite" id="0-1164" name="Foreman" fullname="Foreman" runstate="Runnable" testcasecount="10">
      <test-suite type="TestSuite" id="0-1165" name="Tests" fullname="Foreman.Tests" runstate="Runnable" testcasecount="10">
        <test-suite type="TestSuite" id="0-1166" name="HelpersTest" fullname="Foreman.Tests.HelpersTest" runstate="Runnable" testcasecount="10">
          <test-suite type="TestFixture" id="0-1167" name="AppFamilyUtilitiesTests" fullname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests" runstate="Runnable" testcasecount="10">
            <test-suite type="ParameterizedMethod" id="0-1169" name="GetRoleIdToAppFamilyToADGroupCNTests" fullname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests.GetRoleIdToAppFamilyToADGroupCNTests" classname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests" runstate="Runnable" testcasecount="2">
              <properties>
                <property name="Category" value="AppFamilyUtilitiesTests" />
              </properties>
              <test-case id="0-1090" name="Empty AppFamilyGroupModel list" fullname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests.Empty AppFamilyGroupModel list" methodname="GetRoleIdToAppFamilyToADGroupCNTests" classname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests" runstate="Runnable" seed="195536145">
                <properties />
              </test-case>
              <test-case id="0-1091" name="Three roles and with multiple app families per role" fullname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests.Three roles and with multiple app families per role" methodname="GetRoleIdToAppFamilyToADGroupCNTests" classname="Foreman.Tests.HelpersTest.AppFamilyUtilitiesTests" runstate="Runnable" seed="700917615">
                <properties />
              </test-case>
            </test-suite>
          </test-suite>
        </test-suite>
      </test-suite>
    </test-suite>
  </test-suite>
</test-run>
```

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/nunit/nunit?pullRequest=3612) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/nunit/nunit?pullRequest=3612) before we can accept your contribution.<br/><br/>:x: cjconrad2 [sign now](https://cla.dotnetfoundation.org/nunit/nunit?pullRequest=3612)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/nunit/nunit?pullRequest=3612) it.</sub>

File: src/NUnitFramework/framework/Internal/Tests/TestSuite.cs
Comment (mikkelbu) [inline src/NUnitFramework/framework/Internal/Tests/TestSuite.cs:113]:
```suggestion
            foreach (string key in suite.Properties.Keys)
                foreach (object val in suite.Properties[key])
                    this.Properties.Add(key, val);
```

**Review by mikkelbu — APPROVED** [review]
I think it is a fine way to test that the filtering does not change the properties. Perhaps we should also check that there is some data returned by `Properties.Get(...)`, so that the checks are not successful by accident (i.e. both of them return `null`).

I was going to say that this would only solve #3611 partly as there were some issue about no `TestFixture`, but I think you already solve this in #3172. We can merge this when we get another review.

**Review by rprouse — APPROVED** [review]

**Pull request merged**
